ntbs-service/Pages/ValidationModel.cs

```diff
@@ -3,14 +3,20 @@
 using Microsoft.AspNetCore.Mvc.RazorPages;
 using ntbs_service.Models;
 using ntbs_service.Models.Validations;
-using System.Linq;
 using System.Collections.Generic;
 using Newtonsoft.Json;
+using EFAuditer;
+using System.Threading.Tasks;
 
 namespace ntbs_service.Pages
 {
     public abstract class ValidationModel : PageModel
     {
+        protected async Task OnGetAuditAsync(int notificationId, object model)
+        {
+            await Auditer.AuditReadOperation("NotificationId", notificationId, model);

```

Comment:
Should probably not use the static method here but inject Auditer - see comments in Auditer.cs above.

Reply:
In a way it doesn't feel very nice to have to manually call this when the rest of the library is so automated, but I don't really see a way around this - we obviously wouldn't want to log every database get/select as there will be tons, and really the only things we care about is whether any of the "Get" pages were hit, so hopefully this way makes sense!

Reply:
Yes, I think that this is where we want to log it. I wonder if there's a way to hook into the request lifecycle to make this nicer, so that every GET request to `/Notification/...` gets picked up, before reaching the controllers?
Though we could argue it's for https://airelogic-nis.atlassian.net/browse/NTBS-205 to finalise, unless you'd rather finish it off here! (the boundary between the two is very blurry and already muddled :P)

Reply:
I assume it would be possible to hook into the request pipeline, however we don't really want to only log when someone makes a get request, because at that point we only have the id of the request, and nothing else which would presumably be not very useful? I guess it depends on what exactly they want to know about reads, so best to wait until the further implementation tickets :)

Reply:
Yup, sounds right. Potentially ID is all we'll need, but equally putting int on request might not make sense if the requests fail, so it might have to be on the way out. let's leave that to next ticket :)